lib/chainer/functions/connection/embed_id.rb

```diff
@@ -0,0 +1,50 @@
+module Chainer
+  module Functions
+    module Connection
+      class EmbedIDFunction < Chainer::Function
+        def initialize(ignore_label: nil)
+          @ignore_label = ignore_label
+        end
+
+        def self.embed_id(x, w, ignore_label: nil)
+          self.new(ignore_label: ignore_label).(x, w)
+        end
+
+        def forward_cpu(inputs)
+          (x, w) = inputs
+
+          unless @ignore_label
+            return [Chainer::Utils::Array.take(w, x.to_a, axis: 0)]
```

Comment:
Avoid `to_a` as much as possible for performance. It requires converting NArray into Ruby Array. Conversion and especially memory copying makes slow.